src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -33368,635 +33971,3631 @@ .maxstack  3
+        public void AddMethod_ByRef_01(string refKind)
+        {
+            string source = $$"""
+                using System.Collections;
+                using System.Collections.Generic;
+                class MyCollection<T> : IEnumerable
+                {
+                    private readonly List<T> _list = new();
+                    IEnumerator IEnumerable.GetEnumerator() => _list.GetEnumerator();
+                    public void Add({{refKind}} T t) { _list.Add(t); }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        int x = 1;
+                        int[] y = [2, 3];
+                        MyCollection<int?> z = [x, ..y, null];
+                        z.Report();
+                    }
+                }
+                """;
+            CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3, null], ");
+        }
+
+        [Theory]
+        [InlineData("out")]
+        [InlineData("ref")]
+        public void AddMethod_ByRef_02(string refKind)
+        {
+            string source = $$"""
+                using System.Collections;
+                using System.Collections.Generic;
+                class MyCollection<T> : IEnumerable
+                {
+                    private readonly List<T> _list = new();
+                    IEnumerator IEnumerable.GetEnumerator() => _list.GetEnumerator();
+                    public void Add({{refKind}} T t) { t = default; _list.Add(t); }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        int x = 1;
+                        int[] y = [2, 3];
+                        MyCollection<int?> z = [x, ..y, null];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (15,32): error CS1954: The best overloaded method match 'MyCollection<int?>.Add(ref int?)' for the collection initializer element cannot be used. Collection initializer 'Add' methods cannot have ref or out parameters.
+                //         MyCollection<int?> z = [x, ..y, null];
+                Diagnostic(ErrorCode.ERR_InitializerAddHasParamModifiers, "[x, ..y, null]").WithArguments("MyCollection<int?>.Add(" + refKind + " int?)").WithLocation(15, 32),
+                // (15,32): error CS9215: Collection expression type 'MyCollection<int?>' must have an instance or extension method 'Add' that can be called with a single argument.
+                //         MyCollection<int?> z = [x, ..y, null];
+                Diagnostic(ErrorCode.ERR_CollectionExpressionMissingAdd, "[x, ..y, null]").WithArguments("MyCollection<int?>").WithLocation(15, 32));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("in")]
+        [InlineData("ref readonly")]
```

Comment:
Why does this work for `ref readonly`? I would, at the very least, expect a warning?

Reply:
> Why does this work for `ref readonly`? 

Why do you think it shouldn't?  

> I would, at the very least, expect a warning?

A warning where? Saying what?

Are you referring to: "warning CS9192: Argument 1 should be passed with 'ref' or 'in' keyword" 

Note, we intentionally do not report it for ```MyCollection<C> z = new () {x};``` today because there is no place to add any of the suggested keywords to make the warning to go away.